### PR TITLE
 158124343 Add Version 2 tests

### PR DESCRIPTION
**Pull request by migot01**
## What does this PR do?

This PR adds version 2 test to the App

## How should this be manually tested?
$ git clone https://github.com/migot01/Maintraq/tree/ft-user-request-158194038
$ cd maintraq
Create a virtual environment
pipenv shell
Activate a virtual environment

Install project dependencies
cd in test folder and run pytest

screenshot

image

### CODE REVIEW

File: Pipfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (zmwas) [inline Pipfile.lock]: Put this in a .gitignore

File: .pytest_cache/v/cache/nodeids [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (zmwas) [inline .pytest_cache/v/cache/nodeids]: put this in a gitignore and delete it

File: app/__init__.py [hunk 3]
```diff
@@ -1,11 +1,7 @@
...
-app.config['SECRET_KEY'] = 'thisissecret'
-def create_app(config_name):
-    app = Flask(__name__)
-    
-    app.register_blueprint(views2)
-    return app  
\ No newline at end of file
+app.register_blueprint(views2)
+from config import app_config
+app.config['SECRET_KEY'] = 'thisissecret'
```
Comment (zmwas) [inline app/__init__.py]: use an env variable for this.

File: config.py [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+
+base_dir = os.path.abspath(os.path.dirname(__file__))
+
+try:
+    #connect to database
+    conn = psycopg2.connect("dbname='maintraq_test' user='postgres' host='localhost' password='myadmin01?'")
+    
+    
+except:
+    print("database not connected")
```
Comment (zmwas) [inline config.py]: remove this print statement

File: config.py [hunk 5]
```diff
@@ -1,9 +1,22 @@
+import os
+import psycopg2
+
+base_dir = os.path.abspath(os.path.dirname(__file__))
+
+try:
+    #connect to database
+    conn = psycopg2.connect("dbname='maintraq_test' user='postgres' host='localhost' password='myadmin01?'")
```
Comment (zmwas) [inline config.py]: these credentials should not be here add them in an env file

File: tests/.pytest_cache/v/cache/lastfailed [hunk 6]
```diff
@@ -1,5 +1,11 @@
 {
```
Comment (zmwas) [inline tests/.pytest_cache/v/cache/lastfailed]: gitignore

File: tests/test_login.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import unittest
 from app import app
-from app import create_app
+#from app import create_app
```
Comment (zmwas) [inline tests/test_login.py]: delete this if it is not useful

File: tests/.pytest_cache/v/cache/nodeids [hunk 8]
```diff
@@ -1,16 +1,33 @@
 [
```
Comment (zmwas) [inline tests/.pytest_cache/v/cache/nodeids]: gitignore

File: tests/.pytest_cache/v/cache/lastfailed [hunk 9]
```diff
@@ -1,5 +1,11 @@
 {
+  "test_loginV2.py::LoginuserTestcase::test_bad_request_with_wrong_filds": true,
```
Comment (zmwas) [inline tests/.pytest_cache/v/cache/lastfailed]: gitignore

File: tests/.coverage [hunk 10]
```diff
@@ -0,0 +1 @@
...ews2.py":[1,2,3,5,6,7,8,9,10,11,14,16,20,21,23,46,58,78,108,109,110,47,56,48,55,24,44,132,133,134,141,142,143,148,149,150,176,177,178,185,186,187,194,195,196,204,205,206,214,215,216,61,62,66,68,69,63,81,82,83,84,85,86,90,93,96,97,98,99,100,71,72,75,87,88,94,26,27,31,32,33,34,35,36,37,43,50,54,114,115,116,117,120,121,122,123,124,125,126,138,139,154,156,157,158,159,145,146],"C:\\Maintraq\\app\\models2.py":[1,2,3,4,5,6,8,10,11,19,31,52,53,54,62,12,13,14,15,23,25,36,38,39,40,41,42,44,45,46,47,55,56,57,58,64,65],"C:\\Maintraq\\app\\helpers.py":[1,2,4,5,6,7,8,9,12,14,16,22,32,41,47,57,62,69,74,81,87,92,97,104,42,43,44,17,18,20,58,59,60,61,63,64,65,67,70,71,72],"C:\\Maintraq\\app\\views.py":[1,2,3,4,5,6,7,8,9,11,13,15,16,19,40,63,86,87,20,38,109,110,118,119,144,145,152,153,165,166,43,44,45,47,49,51,52,53,54,55,56,57,58,66,67,68,70,72,73,74,75,76,77,79,80,22,23,27,28,29,30,31,37,90,91,92,94,98,99,100,101,102,103,169,170,171,172,173,174,175,176,50,177,82,83,84,71,69,34,35,156,157,158,159,160,161,33,24,25,26,59,60,61,46,48,95,96,93,113,114,115,116,123,124,126,127,128,129,130,131,133,134,135,136],"C:\\Maintraq\\app\\models.py":[1,3,5,9,18,19,20,23,6,7,21,13,14,15,16,25,27,28,29,30,31,33,34]}}
```
Comment (zmwas) [inline tests/.coverage]: remove this file and add it to a gitignore.

Comment (migot01) [reply tests/.coverage]: thanks, implementing the changes right away

**Pull request merged**